File: elixir/01.exs
defmodule Fuel do
    defp read_input do
        File.read!(Path.expand("../inputs/input01.txt"))
        |> String.split("\n")
        |> Enum.map(fn line -> line |> Integer.parse end)
        |> Enum.map(fn {num, _} -> num end)
    end

    defp calc_fuel(mass) do
        div(mass, 3) - 2
    end

    defp calc_fuel(mass, accum_fuel \\ 0) do
        f = calc_fuel(mass)
        cond do
            f > 0 -> calc_fuel(f, accum_fuel + f)
            true  -> accum_fuel
        end
    end

    def part1 do
        read_input()
        |> Enum.map(&calc_fuel/1)
        |> Enum.sum
    end

    def part2 do
        read_input()
        |> Enum.map(&(calc_fuel(&1, 0)))
        |> Enum.sum
    end
end

File: go/01.go
package main

import (
	"fmt"
	"os"
)

func main() {
	sum := 0
	for i := 0; i < 10; i++ {
		sum += i
	}
	fmt.Println(sum)
}

fp, err := os.Open("../inputs/input01.txt")
modules = fp.read().split("\n").map_to_int

scanner := bufio.NewScanner(file)
scanner.Split(bufio.ScanLines)
var lines []string

for scanner.Scan() {
	lines = append(lines, scanner.Text())
}

file.Close()

func calc_fuel(mass) {
	floor(mass / 3) - 2
}

func part1() {
	modules.map().sum()
}

part1();
